internal/acceptance/kubernetes/kubernetes.go

```diff
@@ -14,95 +14,102 @@
-		WillReturn(fmt.Sprintf(`{
-				"apiVersion": "appstudio.redhat.com/v1alpha1",
-				"kind": "EnterpriseContractPolicy",
-				"metadata": {
-				  "name": "%s",
-				  "namespace": "%s"
-				},
-				"spec": %s
-			  }`, name, ns, os.Expand(specification.Content, func(key string) string {
-			if key == "GITHOST" {
-				return git.Host(ctx)
-			}
-
-			return ""
-		})),
-			map[string]string{"Content-Type": "application/json"},
-			200,
-		))
+func (c ClusterState) Key() any {
+	return clusterStateKey
 }
 
-// KubeConfig returns a valid kubeconfig configuration file in YAML format that
-// points to the stubbed apiserver and uses no authentication
-func KubeConfig(ctx context.Context) (string, error) {
-	server, err := wiremock.Endpoint(ctx)
-	if err != nil {
+func (c ClusterState) Up(ctx context.Context) bool {
+	return c.cluster != nil && c.cluster.Up(ctx)
+}
+
+func (c ClusterState) KubeConfig(ctx context.Context) (string, error) {
+	if err := mustBeUp(ctx, c); err != nil {
 		return "", err
 	}
 
-	cluster := "my-cluster"
-
-	context := "my-context"
-
-	kubeconfig := api.Config{
-		CurrentContext: context,
-		Clusters: map[string]*api.Cluster{
-			cluster: {
-				Server: server,
-			},
-		},
-		Contexts: map[string]*api.Context{
-			context: {
-				Cluster: cluster,
-			},
-		},
+	return c.cluster.KubeConfig(ctx)
+}
+
+type startFunc func(context.Context) (context.Context, types.Cluster, error)
+
+func startAndSetupState(start startFunc) func(context.Context) (context.Context, error) {
+	return func(ctx context.Context) (context.Context, error) {
+		var c = &ClusterState{}
+		ctx, err := testenv.SetupState(ctx, &c)
+		if err != nil {
+			return ctx, err
+		}
+
+		if c.Up(ctx) {
+			return ctx, nil
+		}
+
+		ctx, c.cluster, err = start(ctx)
+
+		return ctx, err
 	}
+}
 
-	b, err := clientcmd.Write(kubeconfig)
-	if err != nil {
-		return "", err
+func mustBeUp(ctx context.Context, c ClusterState) error {
+	if !c.Up(ctx) {
+		return errors.New("cluster has not been started, use `Given a <stub?> cluster running")
 	}
 
-	return string(b), nil
+	return nil
 }
 
-func IsRunning(ctx context.Context) bool {
-	return wiremock.IsRunning(ctx)
+func createNamedPolicy(ctx context.Context, name string, specification *godog.DocString) error {
+	c := testenv.FetchState[ClusterState](ctx)
+
+	if err := mustBeUp(ctx, *c); err != nil {
+		return err
+	}
+
+	return c.cluster.CreateNamedPolicy(ctx, name, specification.Content)
 }
 
-// AddStepsTo adds Gherkin steps to the godog ScenarioContext
+// AddStepsTo adds cluster-related steps to the context
 func AddStepsTo(sc *godog.ScenarioContext) {
-	sc.Step(`^stub apiserver running$`, stubApiserverRunning)
-	sc.Step(`^policy configuration named "([^"]*)" with specification$`, stubPolicy)
+	sc.Step(`^a stub cluster running$`, startAndSetupState(stub.Start))
+	sc.Step(`^policy configuration named "([^"]*)" with specification$`, createNamedPolicy)
+
+	// removes all git repositories from the filesystem
```

Comment:
This comment seems inaccurate

Reply:
That's what you get for copy-pasting, thanks for spotting this!